xdsl/parser/core.py

```diff
@@ -819,24 +850,13 @@ def _parse_generic_operation(self, op_type: type[Operation]) -> Operation:
 
         self.parse_punctuation(":", "function type signature expected")
 
-        func_type_pos = self._current_token.span.start
-
         # Parse function type
         func_type = self.parse_function_type()
 
-        if len(args) != len(func_type.inputs):
-            self.raise_error(
-                f"expected {len(func_type.inputs)} operand types but had {len(args)}",
-                func_type_pos,
-            )
+        operands = self.resolve_operands(args, func_type.inputs)
```

Comment:
It feels better if `resolve_operands` is after all parsing.

Reply:
Done.